### ISSUE TITLE
When compiling, it overwrite files without asking

### ISSUE THREAD

**Issue by Tremebundix** [olivierkes/manuskript#608]
First off, thanks for the great work!

I think I ran into an issue today for version 0.9: when I compile a project and use a file name that already exists, if overwrites the file without asking "Are you sure?". I think that alert should be added to avoid accidentally overwriting the wrong file.

Thanks!

**Comment by worstje** [comment]
I have been unable to reproduce this, nor can I see in the code why it would not ask you to overwrite. (The QFileDialog::Option QFileDialog::DontConfirmOverwrite is not passed to the dialog that lets you select a filename to save.)

I tested this with a fresh project, added some contents to the first chapter and saved it. Then, I went to File->Compile, hit the Export button on the top right, and manually typed in an `export.txt` file.

Then, I repeated the process. I closed the window, reopened the Compile window, selected the `export.txt` I previous saved, and upon confirming the filename, I was asked if I indeed wanted to overwrite that file.

Can you supply a testcase for us which reproduces the issue you have observed? Please also include your operating system and the version of Qt that is installed.

Thank you!

**Comment by Tremebundix** [comment]
OK, I think I found out why you couldn't reproduce it. The bug is a bit more subtle than what I thought, but it is there:

If I follow the exact steps you describe, you are right, it asks me if I want to replace the existing file. However, if instead of calling the file `export.txt`, i call it just `export` (with no extension), Manuskript will save the compiled text as `export.txt` (adding the extension automatically). If there is already a file called `export.txt`, it will overwrite it without prompting.

The same happens with other file formats (e.g., docx).

My OS is Ubuntu 16.04.

**Comment by worstje** [comment]
Thank you for explaining how to trigger the issue.

Based on your description and a quick glance through the source, I know where the problem is at. Due to some inherent limitations in Qt and/or the host OS regarding default suffixes (extensions) and filters, Manuskript tries to be a bit too helpful by adjusting the filename chosen through the dialog to include the desired extension, inadvertently circumventing the check done by the dialog itself.

Suffice to say, that is not acceptable. The filename selected in the dialog by the user should be set in stone and not silently change in the background. Differing from established behaviour by the operating system is even more-so a definite no-no.

Since we don't appear to really need to circumvent the limitation to begin with, I will probably provide a PR later today that rips out the bandaid and fully relies on the OS for the matter of determining the filename. This could possibly mean some people will try to export files with the incorrect extension (I can't quite test whether or not that is the case at this point before even touching the code), but if that is the case, then they would make the same mistake anywhere else they'd save a file on their computer, too.

The only alternative is a (far uglier) band-aid to check if the new filename conflicts, and if it does not, send the user back to the dialog to choose a filename. If given the option, I'd rather make the code simpler than messier. 😄

**Comment by worstje** [comment]
@Tremebundix : are you in a position to test the above Pull Request to verify whether the fix works as expected for you on Ubuntu?

If you are unfamiliar on how to do test this, you can find instructions here: [Run Manuskript from Source Code on Linux](https://github.com/olivierkes/manuskript/wiki/Run-Manuskript-from-Source-Code-on-Linux). You'll want to use option B that uses `git`.

That only gets you the main develop branch, however.

You will have to check out the pull request into its own branch afterwards as follows:
> `git fetch origin pull/613/head:pr-613`

And then switch to this new branch as follows:
> `git checkout pr-613`

At that point, you can run manuskript (`bin/manuskript`) from there as usual to test out the pull request to see if it is indeed fixed. 😄

**Comment by Tremebundix** [comment]
I just tried, but now when I press the Export button, nothing happens. And I get this output in the console:

```
manuskript$ bin/manuskript 
Debug: Web rendering engine used: QWebView
Running manuskript version 0.9.0.
Note: No translator found or loaded from system locale for locale en_US.
Last accessed directory "/home/cristian/Downloads" loaded.
Project test9.msk saved.
Traceback (most recent call last):
  File "/home/cristian/Downloads/manuskript/bin/../manuskript/ui/exporters/exporter.py", line 107, in export
    F.export(self.settingsWidget)
  File "/home/cristian/Downloads/manuskript/bin/../manuskript/exporter/manuskript/plainText.py", line 86, in export
    filename = self.getExportFilename(settingsWidget)
  File "/home/cristian/Downloads/manuskript/bin/../manuskript/exporter/manuskript/plainText.py", line 72, in getExportFilename
    defaultSuffix=self.exportDefaultSuffix)
  File "/home/cristian/Downloads/manuskript/bin/../manuskript/functions/__init__.py", line 402, in getSaveFileNameWithSuffix
    dialog.setSupportedSchemes(("file",))
AttributeError: 'QFileDialog' object has no attribute 'setSupportedSchemes'
```

**Comment by worstje** [comment]
I expected something to maybe silently not work, but how can an attribute be missing?! 😖 

If you happen to be around, can you share with me the versions of **Qt5** and **PyQt5** that you have installed? Based on some source code spelunking (because changelogs aren't all that), I think `setSupportedSchemes()` is only a Qt5.6 thing.

(Ugh. Why can't Qt5 just have a nice commit history or better yet, complete documentation?!)

I'll dig further into this and might even set up my linux VM to test for myself if push comes to shove.

**Comment by worstje** [comment]
It is hopefully fixed now. I did some digital archeology (it took me more effort than I'd like to admit to find the first releases of PyQt 5.0 and Qt 5.0) but I believe I have implemented the correct fix now.

@Tremebundix : Can you test it out once more? `git pull` should suffice to update to the latest version of the PR.

**Comment by gedakc** [comment]
This issue has been tested as fixed with PR #613 for Qt versions 5.7 and up.
